ush/python/pygfs/task/atm_analysis.py

```diff
@@ -116,64 +169,51 @@ def initialize(self: Analysis) -> None:
         FileHandler({'mkdir': newdirs}).sync()
 
     @logit(logger)
-    def variational(self: Analysis) -> None:
-
-        chdir(self.task_config.DATA)
+    def execute(self, aprun_cmd: str, jedi_args: Optional[str] = None) -> None:
+        """Run JEDI executable
 
-        exec_cmd = Executable(self.task_config.APRUN_ATMANLVAR)
-        exec_name = os.path.join(self.task_config.DATA, 'gdas.x')
-        exec_cmd.add_default_arg(exec_name)
-        exec_cmd.add_default_arg('fv3jedi')
-        exec_cmd.add_default_arg('variational')
-        exec_cmd.add_default_arg(self.task_config.jedi_yaml)
+        This method will run JEDI executables for the global atm analysis
 
-        try:
-            logger.debug(f"Executing {exec_cmd}")
-            exec_cmd()
-        except OSError:
-            raise OSError(f"Failed to execute {exec_cmd}")
-        except Exception:
-            raise WorkflowException(f"An error occured during execution of {exec_cmd}")
+        Parameters
+        ----------
+        aprun_cmd : str
+           Run command for JEDI application on HPC system
+        jedi_args : List
+           List of additional optional arguments for JEDI application
 
-        pass
-
-    @logit(logger)
-    def init_fv3_increment(self: Analysis) -> None:
-        # Setup JEDI YAML file
-        self.task_config.jedi_yaml = os.path.join(self.task_config.DATA,
-                                                  f"{self.task_config.JCB_ALGO}.yaml")
-        save_as_yaml(self.get_jedi_config(self.task_config.JCB_ALGO), self.task_config.jedi_yaml)
+        Returns
+        ----------
+        None
+        """
+        super().execute()
```

Comment:
```suggestion
```
`super().execute()` has nothing in it.  It doesn't need to be called.

Reply:
@aerorahul  This reason I add those is as a reminder to the reader that this is a child class with conventions about which methods it should have.

Reply:
That makes sense, but it is not necessary.  It is known that it is a child of `Task` at declaration `class AtmAnalysis(Task)`.
If `Task` was an `ABC`, the user would get a strongly worded error reminder.

Reply:
Also, you expose this class to upstream `Task.py` changes.  If someone adds something that is not desired in `Task.py`, calling `super().execute()` will inherit it and will cause issues here.  That is not safe.

Reply:
True, but the power of inheritance is that we can change the parent class and have those changes propagate down to the children.

Reply:
Agreed.  However, task.py is not meant to provide that with the exception of `__init__.py`.  
`initialize`, `finalize` may have very different meanings for tasks in an application.
Abstract Base Classes provide the functionality for inheritance, while allowing for variety and being too prescriptive.

Reply:
Just saw your other comment/suggested change on ABCs. I need to do some research on that. Thanks for the feedback.